src/code/code_800F7260.c

```diff
@@ -1,40 +1,639 @@
+    SoundBankEntry* entry;
+    u8 bankIndex = gSoundBanks[SFX_BANK(sfxId)][0].next;
+    u8 bankIndex2 = 0;
+    SoundBankEntry sp50;
+
+    while (bankIndex != 0xFF) {
+        entry = &gSoundBanks[SFX_BANK(sfxId)][bankIndex];
+        if (entry->posX == &pos->x && entry->unk_28 == sfxId) {
+            if (entry->unk_2A >= 3) {
+                func_800E5B20(0x06020000 | ((entry->unk_2E & 0xFF) << 8), 0);
+            }
+            if (entry->unk_2A != 0) {
+                func_800F7B54(SFX_BANK(sfxId), bankIndex);
+            }
+            bankIndex = 0xFF;
+        } else {
+            bankIndex2 = bankIndex;
+        }
+        if (bankIndex != 0xFF) {
+            bankIndex = gSoundBanks[SFX_BANK(sfxId)][bankIndex2].next;
+        }
+    }
+    sp50.posX = &pos->x;
+    sp50.unk_28 = sfxId;
+    func_800F74E0(3, &sp50);
+}
+
+void func_800F8BA0(u8 arg0, u16 sfxId) {
+    SoundBankEntry* entry;
+    u8 bankIndex = gSoundBanks[SFX_BANK(sfxId)][0].next;
+    u8 bankIndex2 = 0;
+    SoundBankEntry sp58;
+
+    while (bankIndex != 0xFF) {
+        entry = &gSoundBanks[SFX_BANK(sfxId)][bankIndex];
+        if (entry->unk_C == arg0 && entry->unk_28 == sfxId) {
+            if (entry->unk_2A >= 3) {
+                func_800E5B20(0x06020000 | ((entry->unk_2E & 0xFF) << 8), 0);
+            }
+            if (entry->unk_2A != 0) {
+                func_800F7B54(SFX_BANK(sfxId), bankIndex);
+            }
+        } else {
+            bankIndex2 = bankIndex;
+        }
+        if (bankIndex != 0xFF) {
+            bankIndex = gSoundBanks[SFX_BANK(sfxId)][bankIndex2].next;
+        }
+    }
+    sp58.unk_C = arg0;
+    sp58.unk_28 = sfxId;
+    func_800F74E0(4, &sp58);
+}
+
+void func_800F8D04(u32 sfxId) {
+    SoundBankEntry* entry;
+    u8 bankIndex = gSoundBanks[SFX_BANK(sfxId)][0].next;
+    u8 bankIndex2 = 0;
+    SoundBankEntry sp50;
+
+    while (bankIndex != 0xFF) {
+        entry = &gSoundBanks[SFX_BANK(sfxId)][bankIndex];
+        if (entry->unk_28 == sfxId) {
+            if (entry->unk_2A >= 3) {
+                func_800E5B20(0x06020000 | ((entry->unk_2E & 0xFF) << 8), 0);
+            }
+            if (entry->unk_2A != 0) {
+                func_800F7B54(SFX_BANK(sfxId), bankIndex);
+            }
+        } else {
+            bankIndex2 = bankIndex;
+        }
+        bankIndex = gSoundBanks[SFX_BANK(sfxId)][bankIndex2].next;
+    }
+    sp50.unk_28 = sfxId;
+    func_800F74E0(5, &sp50);
+}
+
+void func_800F8E3C(void) {
+    while (D_801333A0 != D_801333A4) {
+        func_800F7680();
+        D_801333A4++;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_800F7260/func_800F87A0.s")
+void func_800F8EA0(u8 arg0, u8 arg1, u16 arg2) {
+    f32 temp_f0;
+    Struct_800F8EA0* temp_v0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_800F7260/func_800F8884.s")
+    if (arg2 == 0) {
+        arg2++;
+    }
+    temp_v0 = &D_8016E270[arg0];
+    temp_f0 = arg1 / 127.0f;
+    temp_v0->unk_4 = temp_f0;
```

Comment:
is this temp_f0 temp really mandatory (and temp_v0?)

Reply:
Both were removable it turns out